cmd/internal/versionfile/version_file.go

```diff
@@ -0,0 +1,84 @@
+package versionfile
+
+import (
+	"fmt"
+	"github.com/kumahq/ci-tools/cmd/internal/github"
+	"github.com/kumahq/ci-tools/cmd/internal/version"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func Dev(edition string) VersionEntry {
+	return VersionEntry{
+		Release: "dev",
+		Edition: edition,
+		Version: "preview",
+		Branch:  "master",
+	}
+}
+
+type VersionEntry struct {
+	Edition       string `yaml:"edition"`
+	Version       string `yaml:"version"`
+	Release       string `yaml:"release"`
+	Latest        bool   `yaml:"latest,omitempty"`
+	ReleaseDate   string `yaml:"releaseDate,omitempty"`
+	EndOfLifeDate string `yaml:"endOfLifeDate,omitempty"`
+	Branch        string `yaml:"branch"`
+}
+
+func (v VersionEntry) Less(o VersionEntry) bool {
+	majorI, minorI, _ := version.MustSplitSemVer(strings.ReplaceAll(v.Version, "x", "0"))
+	majorJ, minorJ, _ := version.MustSplitSemVer(strings.ReplaceAll(o.Version, "x", "0"))
+	if majorI == majorJ {
+		return minorI < minorJ
+	}
+	return majorI < majorJ
```

Comment:
don't we care about `patch`? also same thing, a dedicated semver lib would take care of that

Reply:
If you look above the version is `X.Y.x` we replace x by 0 to have the parsing work. So the patch is always 0 thus we ignore it.